include/ensmallen_bits/problems/ackley_function_impl.hpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file ackley_function_impl.hpp
+ * @author Suryoday Basak
+ *
+ * Implementation of the Ackley function.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_PROBLEMS_ACKLEY_FUNCTION_IMPL_HPP
+#define ENSMALLEN_PROBLEMS_ACKLEY_FUNCTION_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "ackley_function.hpp"
+
+namespace ens {
+namespace test {
+
+inline AckleyFunction::AckleyFunction(const double epsilon) : epsilon(epsilon)
+{ /* Nothing to do here */ }
+
+inline void AckleyFunction::Shuffle() { /* Nothing to do here */ }
+
+inline double AckleyFunction::Evaluate(const arma::mat& coordinates,
+                                       const size_t /* begin */,
+                                       const size_t /* batchSize */) const
+{
+  // For convenience; we assume these temporaries will be optimized out.
+  const double x1 = coordinates(0);
+  const double x2 = coordinates(1);
+
+  const double objective = -20 * std::exp(-0.2 * std::sqrt(0.5 * (std::pow(x1, 2) + 
```

Comment:
I'm wondering if we should make ``c`` configurable and use a default value of ``2 pi ``as done here: https://www.sfu.ca/~ssurjano/ackley.html. Let me know what you think.

Reply:
Okay I'll take care of that. I'll include c in the arguments' list